SignalMessaging/ViewModels/ContactShareViewModel.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+@objc
+public class ContactShareViewModel: NSObject {
+
+    public let dbRecord: OWSContact
+    public let avatarImage: UIImage?
+
+    public required init(contactShareRecord: OWSContact, avatarImage: UIImage?) {
+        self.dbRecord = contactShareRecord
+        self.avatarImage = avatarImage
+    }
+
+    public convenience init(contactShareRecord: OWSContact, transaction: YapDatabaseReadTransaction) {
+        if let avatarAttachment = contactShareRecord.avatarAttachment(with: transaction) as? TSAttachmentStream {
+            self.init(contactShareRecord: contactShareRecord, avatarImage: avatarAttachment.image())
+        } else {
+            self.init(contactShareRecord: contactShareRecord, avatarImage: nil)
+        }
+    }
+
+    public func getAvatarImage(diameter: CGFloat, contactsManager: OWSContactsManager) -> UIImage {
+        if let avatarImage = avatarImage {
+            return avatarImage
+        }
+
+        // TODO: What's the best colorSeed value to use?
+        let avatarBuilder = OWSContactAvatarBuilder(nonSignalName: displayName,
+                                                    colorSeed: displayName,
+                                                    diameter: UInt(diameter),
+                                                    contactsManager: contactsManager)
+        return avatarBuilder.build()
```

Comment:
Should we cache this result in `avatarImage`?

Reply:
The avatar builder has built in caching per resolution which ensures our generated initials (text) avatars, which are especially sensitive to compression/resizing artifacts are to-spec, so I'm inclined to leave this as is and rely on the avatar builder cache.